### PR TITLE
Add models for case transformation

### PR DESCRIPTION
**Pull request by rkondratowicz**
Based on swagger spec from:
https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1064666568#TechnicalprerequisitesandinformationforServiceon-boardingwithPhase2Bulkscanning-2.4.3.Exceptionrecordtransformationendpoint

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/model/in/ExceptionRecord.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    public final JourneyClassification journeyClassification;
+    public final LocalDateTime deliveryDate;
+    public final LocalDateTime openingDate;
+    public final List<ScannedDocument> scannedDocuments;
+    public final List<OcrDataField> ocrDataFields;
+
+    public ExceptionRecord(
+        @JsonProperty("case_type_id") String caseTypeId,
+        @JsonProperty("po_box") String poBox,
+        @JsonProperty("jurisdiction") String jurisdiction,
```
Comment (aliveni) [inline src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/model/in/ExceptionRecord.java]: in tech spec, we have this field as `po_box_jurisdiction`. are you going to change it in tech spec?

File: src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/model/in/ExceptionRecord.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        @JsonProperty("journey_classification") JourneyClassification journeyClassification,
+        @JsonProperty("delivery_date") LocalDateTime deliveryDate,
+        @JsonProperty("opening_date") LocalDateTime openingDate,
+        @JsonProperty("scanned_document") List<ScannedDocument> scannedDocuments,
```
Comment (aliveni) [inline src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/model/in/ExceptionRecord.java]: JsonProperty value should be `scanned_documents`?

File: src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/model/in/ExceptionRecord.java (hunk 1 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/model/in/ExceptionRecord.java]: I've assumed it was a bug in the spec. But apparently not, just checked with @lgonczar 
Updated code.

File: src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/model/in/ExceptionRecord.java (hunk 2 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/model/in/ExceptionRecord.java]: fixed

**Comment by aliveni** [comment]
just realized there is a model class missing for `InvalidExceptionRecordResponse` in case of 422 response code

**Pull request merged**